### PR TITLE
fix(core): Mark mptv2 items as inherited

### PR DESCRIPTION
**Pull request by louisjimenez**
Marks inherited items (triggers, notifications, parameters) as "inherited" so that the UI can distinguish between what is on the pipeline config and editable versus what is read only from the template.
https://github.com/spinnaker/spinnaker/issues/4451

### CODE REVIEW

**Comment by louisjimenez** [comment]
Related PR for deck: https://github.com/spinnaker/deck/pull/7102

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java [hunk 1]
```diff
@@ -110,8 +111,25 @@ private void addTriggers(
       pipeline.put(
           "triggers",
           TemplateMerge.mergeDistinct(
-              (List<HashMap<String, Object>>) template.getPipeline().get("triggers"),
+              markInherited(getTemplateCollection(template, "triggers")),
               configuration.getTriggers()));
     }
   }
+
+  private static List<HashMap<String, Object>> markInherited(
```
Comment (maggieneterval) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]:
```suggestion
  private static List<Map<String, Object>> markInherited(
```

Comment (maggieneterval) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: I think we typically want to have return types be interfaces unless there's a good reason for them to be implementations

Comment (maggieneterval) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: Separately, I feel like the name `markInherited` incorrectly implies that we are mutating the input. I don't have a good alternative name though...

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java [hunk 2]
```diff
@@ -110,8 +111,25 @@ private void addTriggers(
...
+    return templateCollection.stream()
+        .map(
+            templateItem -> {
+              templateItem.put("inherited", true);
+              return templateItem;
+            })
+        .collect(Collectors.toList());
+  }
+
+  private static List<HashMap<String, Object>> getTemplateCollection(
```
Comment (maggieneterval) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]:
```suggestion
  private static List<Map<String, Object>> getTemplateCollection(
```

**Comment by maggieneterval** [comment]
Adding @duftler as well since Eric is OOO :smile_cat:

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java (hunk 1 above)
Comment (maggieneterval) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: `getInheritanceDesignatedTemplateCollections`? Probably too verbose, ignore me :see_no_evil:

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java (hunk 2 above)
Comment (louisjimenez) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: Ah good call out, definitely, thanks!

Comment (louisjimenez) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: hmmm now I recall why I did this, configuration.getParameters returns a HashMap and mergeDistinct expects both arguments to be of the same type

Comment (maggieneterval) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: ooooh gotcha :+1:

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java (hunk 1 above)
Comment (louisjimenez) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: 👍 I agree that markInherited is a bit confusing. I am renaming and refactoring to make it clearer that it mutates with an `inherited` key

Comment (maggieneterval) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java]: Nice LGTM :100:

**Comment by louisjimenez** [comment]
@spinnakerbot cherry-pick 1.14

**Pull request merged**